### DISCUSSION TITLE
Yi1080p Home FTP files are in different timezone

### DISCUSSION THREAD

**Discussion by blfriedman** [roleoroleo/yi-hack-Allwinner-v2#440] [General]
I have just purchased a new Yi 1080p Home camera, Model YYS.2016 manufacture date 2022.01  Serial No: RFUS,
firmware base 9.0.35.00_202111291330

I am using the Allwinner-v2 y211ga for the RFUS version 0.2.2.

I have configured the timezone to "CST6CDT,M3.2.0,M11.1.0" for Chicago, IL, USA.

I notice that the files are five hours in the future, and upon logging in to the system, it is showing the response to the "date" command also five hours in the future. 

Rebooting the camera does not solve the problem.

I saw this behavior on the recent version of the Yi-Hack for M*Star - as I have many of Yi 1080p Home of the M*Star variety.  It seemed to disappear after a day of use for some reason.

Compared to the M*Star variety of Yi-Hack, when I issue the "date" command at the terminal, M*Star is providing correct local time, but the Allwinner Yi-Hack is not (currently).  I suspect this is the reason the ftp file dates (by directory name) are off the same amount of time.

**Comment by blfriedman** [comment]
Also, note that the time shown on the status webpage for local time is correct local time, and the time in the /tmp/ftppush.log logfile is correct local time, but the date command on linux is in the future.

```
root@Yi_46A451:~# cat /tmp/ftppush.log

2022-05-23 [19-45-37] [INFO] === SERVICE START ===
2022-05-23 [19-45-38] [INFO] checkFiles
2022-05-23 [19-46-23] [INFO] checkFiles
root@Yi_46A451:~# date
Tue May 24 00:47:08 GMT 2022
```

It appears that the TZ environment variable is not getting set - as this command works right:

```
root@Yi_46A451:~# TZ="CST6CDT,M3.2.0,M11.1.0" date
Mon May 23 19:53:29 CDT 2022
root@Yi_46A451:~# date
Tue May 24 00:53:31 GMT 2022
root@Yi_46A451:~# date -u
Tue May 24 00:53:36 UTC 2022
```

and if exported, the date command is correct

```
root@Yi_46A451:~# export TZ=CST6CDT,M3.2.0,M11.1.0
root@Yi_46A451:~# date
Mon May 23 19:55:22 CDT 2022
```
Near the end of script/system.sh there is an "unset TZ".  Commenting this out does not solve the issue.

**Comment by roleoroleo** [comment]
TZ variable is not set in the shell you are using.
This is normal because the rootfs is read only.
As you can see here:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/blob/master/src/static/static/yi-hack/script/system.sh
at line 175, the variable is exported before running the custom processes.

Try to source the env:
`. /tmp/sd/yi-hack/script/env.sh`
